shell/server/pre-meteor.js

```diff
@@ -90,7 +96,15 @@ Meteor.startup(function () {
       publicIdPromise = Promise.resolve(id);
     } else {
       // Not a wildcard host. Perhaps it is a custom host.
-      publicIdPromise = lookupPublicIdFromDns(hostname);
+
+      // Skip DNS TXT record resolutions for the major pseudo-TLDs, and avoid
+      // returning any result which might indicate that there's actually a
+      // server listening here.
+      if (hasPseudoTLD(hostname)) {
+        return;
+      } else {
+        publicIdPromise = lookupPublicIdFromDns(hostname);
```

Comment:
I have a feeling that the logic here, where for certain "TLDs" we carefully do not do the DNS lookup, would be better in the `lookupPublicIdFromDns` function, though @kentonv or @jparyani might have another opinion.

It seems to me that it's the `lookupPublicIdFromDns` function that is the one whose actions are being adjusted.